source/integrations/mongoose-get-started.txt

```diff
@@ -0,0 +1,717 @@
+            :end-before: end-create-user-improper-email
+            :emphasize-lines: 3
+
+         .. output::
+            :language: console
+            :visible: false
+      
+            Error: Invalid email format
+      
+      If you correct the email address, you can see that the error is not thrown.
+
+      Besides the ``pre()`` middleware function, Mongoose also offers a
+      ``post()`` middleware function. For more information about middleware, see
+      the `Middleware <https://mongoosejs.com/docs/middleware.html>`__ page in
+      the Mongoose documentation.
+
+Next Steps
+----------
+
+You now have a sample project that uses Mongoose to perform CRUD operations on a
+MongoDB collection. From here, you can choose to build on this project with more
+complex queries or document validation. The following sections includes some
+examples of Mongoose features that you might use to in your application.
+
+Mongoose Custom Setters
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Custom setters modify your data when it is saved, and can be implemented similar
+to validators. Mongoose provides the following custom setters:
+
+- ``lowercase``
+- ``uppercase``
+- ``trim`` 
+
+The following example lowercases the characters in the ``blog.slug`` field:
+
+.. code-block:: javascript
+   :emphasize-lines: 7
+
+   const blogSchema = new Schema({
+      ...
+      slug:  {
+         type: String,
+         required: true,
+         minLength: 4,
+         lowercase: true,
+      },
+      ...
+   });
+
+For more information, see the `SchemaStringOptions
+<https://mongoosejs.com/docs/api/schemastringoptions.html#SchemaStringOptions.prototype.lowercase>`__
+page in the Mongoose API documentation.
+
+Helper Methods
+~~~~~~~~~~~~~~
+
+Mongoose includes several helper methods that are abstracted from regular
+MongoDB methods. In this section, you can find examples of some of these
+methods. These methods are not used specifically in this tutorial, but they are
+helpful to reference when getting started with Mongoose.
+
+exists()
+~~~~~~~~
+
+The ``exists()`` method returns either ``null`` or the first document that
+matches the provided query. The following is an example of matching an article
+based on the ``author`` field:
+
+.. code-block:: javascript
+
+   const blog = await Blog.exists({ author: 'Jess Garcia' });
+   console.log(blog);
+
+For more information, see the `Model.exists()
+<https://mongoosejs.com/docs/api/model.html#Model.exists()>`__ section of the
+Mongoose API documentation.
+
+where()
+~~~~~~~
+
+The ``where()`` method allows you to chain and build complex queries. 
+
+The following is an example of performing a find operation by using
+``findOne()`` with a plain query object and the equivalent
+query with the ``where()`` method:
+
+.. code-block:: javascript
+
+   const blogFind = await Blog.findOne({ author: "Jess Garcia" });
+   console.log(blogFind);
+
+   const blogWhere = await Blog.where("author").equals("Jess Garcia");
```

Comment:
While this code does work as written, I wanted to repeat my previous comment that `Blog.find().where('author').equals('Jess Garcia')` is more idiomatic.